File: core/src/main/java/no/nav/pensjon/vtp/core/annotations/SoapService.java
package no.nav.pensjon.vtp.core.annotations;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.stereotype.Component;

/**
 * Indicates that the annotated class is a Soap WebService that should be
 * component scanned and registered with the given <i>path</i>.
 */
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Component
public @interface SoapService {
    String[] path();
}

File: core/src/main/kotlin/no/nav/pensjon/vtp/core/annotations/SoapService.kt
package no.nav.pensjon.vtp.core.annotations

import org.springframework.stereotype.Component
import kotlin.annotation.AnnotationTarget.ANNOTATION_CLASS
import kotlin.annotation.AnnotationTarget.CLASS

/**
 * Indicates that the annotated class is a Soap WebService that should be
 * component scanned and registered with the given *path*.
 */
@Target(ANNOTATION_CLASS, CLASS)
@Retention
@MustBeDocumented
@Component
annotation class SoapService(
    val path: Array<String>
)
